### DISCUSSION TITLE
Very interesting application of the actor model to design systems in real time in collaboration with stakeholders

### DISCUSSION THREAD

**Discussion by bard** [statelyai/xstate#2215] [General]
Video: [Ultra Agile Development](https://www.youtube.com/watch?v=xmHSu5nAFEM)
Post: https://pettergraff.blogspot.com/2015/04/it-all-about-iterations.html

Summary of the video:

1. Developers and stakeholders/domain experts sit at the same table
2. A developer sketches (in high-level code) a system where actors represent various parties in a process (in the example: Hospital, Supplier, and Distributor negotiating the purchase and delivery of equipment)
3. A web service generates on the fly UIs for each actor with widgets that allow sending the messages supported by the actor
4. Stakeholders/domain experts each impersonate the actor and try out to accomplish the various sub-goals of the process (send a price quote, accept the price quote, etc)
5. The developer listens to feedback and refines the system in real time, adding states and messages to support the process envisioned by the stakeholders/domain experts; the web service updates the web UIs
6. Repeat

**Comment by tomByrer** [comment]
TY for reference!  Reminds me of a Javascript Jabber show I hears a few weeks ago about the guest promoting the idea of FSM to build micorservices; the states can be built by or with the stakeholders.
